src/uct/rocm/base/rocm_base.c

```diff
@@ -199,6 +199,30 @@ ucs_status_t uct_rocm_base_detect_memory_type(uct_md_h md, const void *addr,
     return UCS_ERR_INVALID_ADDR;
 }
 
+ucs_status_t uct_rocm_base_mem_query(uct_md_h md, const void *addr,
+                                     size_t length,
+                                     uct_md_mem_attr_t *mem_attr_p)
+{
+    ucs_status_t status;
+
+    mem_attr_p->field_mask = 0;
+
+    if ((addr == NULL) || (length == 0)) {
+        return UCS_ERR_INVALID_PARAM;
+    }
+
+    if (mem_attr_p->field_mask |= UCT_MD_MEM_ATTR_FIELD_MEM_TYPE) {
```

Comment:
why need this `if`?
I think it should be:
```
ucs_status_t uct_rocm_base_mem_query(uct_md_h md, const void *addr,
                                     size_t length,
                                     uct_md_mem_attr_t *mem_attr_p)
{
    uint64_t field_mask = mem_attr_p->field_mask;
    ucs_status_t status;

    mem_attr_p->field_mask = 0;

    if ((addr == NULL) || (length == 0)) {
        return UCS_ERR_INVALID_PARAM;
    }

    if (field_mask & UCT_MD_MEM_ATTR_FIELD_MEM_TYPE) {
        status = uct_rocm_base_detect_memory_type(md, addr, length,
                                                  &mem_attr_p->mem_type);
        if (UCS_OK != status) {
            ucs_warn("unable to query mem_type for %p", (void*) addr);
            return status;
        }
    }

    return UCS_OK;
}
```

if callers asks for `SYS_DEV | MEM_TYPE`, it returns `MEM_TYPE` only in RoCM

Reply:
Thanks for the correction @dmitrygx I've handled this by returning an error if user requests SYS_DEV on rocm md as it is unsupported today.

Reply:
@Akshay-Venkatesh thank you for the explanation!